File: Feladatok/ágazati szakmai érettségi/2017_may/Helsinki1952.cs
using System;
using System.Linq;
using System.IO;
 
namespace ConsoleApplication1{
    class Helsinki1952 {

        public static void Main(String[] args) {
            var helyezesek = File.ReadLines("helsinki.txt")
    						     .Select(line => new Helyezes(line))
    						     .ToArray();
    	
    	    Console.WriteLine("3.Feladat\nPontszerző helyezések száma: " + helyezesek.Length);

            var aranyak = helyezesek.Where(k => k.helyezes == 1).Count();
            var ezustok = helyezesek.Where(k => k.helyezes == 2).Count();
            var bronzok = helyezesek.Where(k => k.helyezes == 3).Count();

            Console.WriteLine("4.Feladat\nAranyak: " + aranyak + ", ezustok: " + ezustok + ", bronzok: " + bronzok + ", összesen: " + (aranyak + ezustok + bronzok));
            Console.WriteLine("5.Feladat\nPontok száma: " + helyezesek.Select(k => k.PontCalc()).Sum());
    	
    	    var uszas = helyezesek.Where(k => k.helyezes <= 3)
                                  .Where(k => k.sportag == "uszas")
                                  .Count();

            var torna = helyezesek.Where(k => k.helyezes <= 3)
                                  .Where(k => k.sportag == "torna")
                                  .Count();

            Console.WriteLine("6.Feladat");
            Console.WriteLine(uszas == torna? "Egyenlőek" : (torna > uszas) ? "Torna több" : "Úszás több");
    	    File.WriteAllLines("helsinki2.txt", helyezesek.Select(k => k.helyezes + " " + k.PontCalc() + " " + k.sportag.Replace("kajakkenu", "kajak-kenu")).ToArray());
    
            var max = helyezesek.OrderByDescending(k => k.sportolokSzama).First();
            Console.WriteLine("Helyezés: " + max.helyezes + ", sportág: " + max.sportag + ", szám: " + max.versenyszam + ", sportolók: " + max.sportolokSzama);
            Console.Read();
        }

        struct Helyezes {
            public int helyezes, sportolokSzama;
            public String sportag, versenyszam;

            public Helyezes(String line) {
                String[] split = line.Split(' ');
                helyezes = int.Parse(split[0]);
                sportolokSzama = int.Parse(split[1]);
                sportag = split[2];
                versenyszam = split[3];
            }

            public int PontCalc() => helyezes == 1 ? 7 : 7 - helyezes;
        }
    }
}

File: Feladatok/ágazati szakmai érettségi/2017_may/Helsinki1952.java
import static java.nio.charset.StandardCharsets.*;

import java.nio.file.*;
import java.util.*;
import java.util.stream.*;

public class Helsinki1952 {
	
    public static void main(String[] args) throws Exception{
    	var helyezesek = Files.lines(Paths.get("helsinki.txt"), ISO_8859_1)
    						  .map(Helyezes::new)
    						  .toArray(Helyezes[]::new);
    	
    	System.out.println("3.Feladat: Pontszerzõ helyezések száma: " + helyezesek.length);
    	
    	var aranyak = Arrays.stream(helyezesek).filter(k -> k.helyezes == 1).count();
    	var ezustok = Arrays.stream(helyezesek).filter(k -> k.helyezes == 2).count();
    	var bronzok = Arrays.stream(helyezesek).filter(k -> k.helyezes == 3).count();
    	
    	System.out.println("4.Feladat: Aranyak: " + aranyak + ", ezustok: " + ezustok + ", bronzok: " + bronzok + ", összesen: " + (aranyak + ezustok + bronzok));
    	System.out.println("5.Feladat: Pontok száma: " + Arrays.stream(helyezesek).mapToInt(Helyezes::pontCalc).sum());
    	
    	var uszas = Arrays.stream(helyezesek)
	    				  .filter(k -> k.helyezes <= 3)
	    				  .filter(k -> k.sportag.equals("uszas"))
	    				  .count();
    	
    	var torna = Arrays.stream(helyezesek)
						  .filter(k -> k.helyezes <= 3)
						  .filter(k -> k.sportag.equals("torna"))
						  .count();
    	
    	System.out.println("6.Feladat");
    	System.out.println(uszas == torna ? "Egyenlõek" : (torna > uszas) ? "Torna több" : "Úszás több");
    	
    	Files.write(Paths.get("helsinki2.txt"), Arrays.stream(helyezesek)
    												  .map(k -> k.helyezes + " " + k.pontCalc() + " " + k.sportag.replace("kajakkenu", "kajak-kenu"))
    												  .collect(Collectors.toList()));
    
    	Arrays.stream(helyezesek).max(Comparator.comparingInt(k -> k.sportolokSzama))
    						 	 .ifPresent(k -> System.out.println("Helyezés: " + k.helyezes + ", sportág: " + k.sportag + ", szám: " 
    								 			+ k.versenyszam + ", sportolók: " + k.sportolokSzama));
    }
    
    static class Helyezes{
    	int helyezes, sportolokSzama;
    	String sportag, versenyszam;
    	
    	public Helyezes(String line) {
    		String[] split = line.split(" ");
    		helyezes = Integer.parseInt(split[0]);
    		sportolokSzama = Integer.parseInt(split[1]);
    		sportag = split[2];
    		versenyszam = split[3];
		}
    	
    	public int pontCalc() { return helyezes == 1 ? 7 : 7 - helyezes; }
    }
}